hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -297,16 +297,13 @@ def incorporate_new_buffer(self, buffer):
         if self.shrink_target.buffer.startswith(buffer):
             return False
 
-        if not self.__engine.prescreen_buffer(buffer):
-            return False
-
-        assert sort_key(buffer) <= sort_key(self.shrink_target.buffer)
-        data = ConjectureData.for_buffer(buffer)
-        self.__engine.test_function(data)
-        self.__test_function_cache[buffer] = data
-        return self.incorporate_test_data(data)
+        previous = self.shrink_target
+        self.cached_test_function(buffer)
+        return previous is not self.shrink_target
 
     def incorporate_test_data(self, data):
+        if data is Overrun:
+            return
```

Comment:
Right. Also note that it's only a cache miss in the sense that it falls back to the slower tree based computation - the underlying engine still notes the buffer and is able to detect that it will lead to an overrun.